src/components/editor/editor-window/tool-bar/editor-preferences/editor-preferences.tsx

```diff
@@ -0,0 +1,67 @@
+import CodeMirror, { EditorConfiguration } from 'codemirror'
+import React, { useState, Fragment, useCallback } from 'react'
+import { Button, ListGroup, Form } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { ForkAwesomeIcon } from '../../../../common/fork-awesome/fork-awesome-icon'
+import { CommonModal } from '../../../../common/modals/common-modal'
+import { EditorPreferenceKeymap } from './editor-preference-keymap'
+import { EditorPreferenceTabchar } from './editor-preference-tabchar'
+import { EditorPreferenceTabsize } from './editor-preference-tabsize'
+import { EditorPreferenceTheme } from './editor-preference-theme'
+
+export interface EditorSettingsButtonProps {
+  editor: CodeMirror.Editor
+  preferences: EditorConfiguration
+  onPreferencesChange: (config: EditorConfiguration) => void
+}
+
+export const EditorPreferences: React.FC<EditorSettingsButtonProps> = ({ editor, onPreferencesChange, preferences }) => {
```

Comment:
I don't know if it's a good idea to create 4 preference specific components, that all have similar code. They all are just a dropdown for text. I think a more elegant solution would be to create a generic "select dropdown with label" component and maintain their options and the state manipulation here.

Reply:
This was a work-in-progress state and now finally just uses a single preference-selection component.